services/packages/cargo/index.go

```diff
@@ -239,7 +241,7 @@ func createOrUpdateConfigFile(ctx context.Context, repo *repo_model.Repository,
 		"Initialize Cargo Config",
 		func(t *files_service.TemporaryUploadRepository) error {
 			var b bytes.Buffer
-			err := json.NewEncoder(&b).Encode(BuildConfig(owner))
+			err := json.NewEncoder(&b).Encode(BuildConfig(owner, repo.IsPrivate))
```

Comment:
Same here. Something like:
`setting.Service.RequireSignInView || ctx.Package.Owner.Visibility != structs.VisibleTypePublic || repo.IsPrivate`
Instance requires login or owner is not public or index repo is private.
If the owner is an organization that check may not be enough too.

Reply:
This one is less of an issue because it can be changed manually later on.